### PR TITLE
[CPU] Split layer nspc -> ncsp special case put back.

### PR DESCRIPTION
**Pull request by maxnick**
Special case for split layer (n(d)hwc -> nc(d)hw) put back to fix the performance degradation (see ticket 46414 for details).
Also this PR contains fix for bug 47383.

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@a-sidorova please review

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp [hunk 1]
```diff
@@ -369,6 +399,47 @@ void MKLDNNSplitNode::selectOptimalPrimitiveDescriptor() {
...
+            auto& vecChildSpd = childPtr->getSupportedPrimitiveDescriptors();
+            const auto& outputDesc = supportedPrimitiveDescriptors[indx].getConfig().outConfs[i].desc;
+
+            if (!vecChildSpd.empty()) {
+                int inNum = childEdge->getOutputNum();
+                if (inNum < 0) {
+                    inNum = 0;
+                }
+                bool hasMatchDesc = false;
+                for (auto& child_spd : vecChildSpd) {
```
Comment (a-sidorova) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp:417]: Please use camel coding style everywhere - `childSpd`

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/split.cpp [hunk 2]
```diff
@@ -115,6 +118,28 @@ const std::vector<Precision> netPrecisions = {
         Precision::BF16
 };
 
+INSTANTIATE_TEST_CASE_P(smoke_Split4D_CPU_Nspc2NcspSpetial, SplitLayerCPUTest,
```
Comment (a-sidorova) [inline inference-engine/tests/functional/plugin/cpu/single_layer_tests/split.cpp:121]: Maybe `spec[t]ial`?

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/split.cpp [hunk 3]
```diff
@@ -115,6 +118,28 @@ const std::vector<Precision> netPrecisions = {
...
+                                ::testing::Values(4),
+                                ::testing::Values(1),
+                                ::testing::ValuesIn(netPrecisions),
+                                ::testing::Values(std::vector<size_t>({3, 28, 24, 9})),
+                                ::testing::Values(std::vector<size_t>({})),
+                                ::testing::Values(CommonTestUtils::DEVICE_CPU),
+                                ::testing::Values(planarChannels_4D)),
+                        SplitLayerCPUTest::getTestCaseName);
+
+INSTANTIATE_TEST_CASE_P(smoke_Split5D_CPU_Nspc2NcspSpetial, SplitLayerCPUTest,
```
Comment (a-sidorova) [inline inference-engine/tests/functional/plugin/cpu/single_layer_tests/split.cpp:132]: `Spec[t]ial` too?

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/split.cpp [hunk 4]
```diff
@@ -91,8 +91,11 @@ const auto planar_5D_ref = CPUSpecificParams{{ncdhw}, {ncdhw}, {"ref"}, "ref"};
 const auto planar_4D = CPUSpecificParams{{nchw}, {nchw}, {}, "unknown"};
 const auto planar_5D = CPUSpecificParams{{ncdhw}, {ncdhw}, {}, "unknown"};
 
-const auto planarChannels_4D = CPUSpecificParams{{nhwc}, {nhwc}, {}, "ref"};
-const auto planarChannels_5D = CPUSpecificParams{{ndhwc}, {ndhwc}, {}, "ref"};
+const auto perChannels_4D = CPUSpecificParams{{nhwc}, {nhwc}, {}, "ref"};
+const auto perChannels_5D = CPUSpecificParams{{ndhwc}, {ndhwc}, {}, "ref"};
+
+const auto planarChannels_4D = CPUSpecificParams{{nhwc}, {nchw}, {}, "ref"};
```
Comment (a-sidorova) [inline inference-engine/tests/functional/plugin/cpu/single_layer_tests/split.cpp:97]: Maybe `perChannelsToPlanar_4D` ?
I think that `planarChannels_4D` doesn't say anything

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp [hunk 5]
```diff
@@ -223,6 +223,16 @@ void MKLDNNSplitNode::initSupportedPrimitiveDescriptors() {
         }
         supportedPrimitiveDescriptors.emplace_back(config, impl_desc_type::unknown, outFormats);
     }
+
```
[the remark this answers is not in the document]
Comment (maxnick) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp:226]: Fixed ca721fa.

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp (hunk 1 above)
Comment (maxnick) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp]: Fixed ca721fa. Thanks.

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/split.cpp (hunk 2 above)
Comment (maxnick) [reply inference-engine/tests/functional/plugin/cpu/single_layer_tests/split.cpp]: Yes, indeed. Fixed ca721fa.

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/split.cpp (hunk 3 above)
Comment (maxnick) [reply inference-engine/tests/functional/plugin/cpu/single_layer_tests/split.cpp]: Fixed ca721fa.

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/split.cpp (hunk 4 above)
Comment (maxnick) [reply inference-engine/tests/functional/plugin/cpu/single_layer_tests/split.cpp]: Yes, certainly. Done ca721fa.

File: inference-engine/tests_deprecated/unit/engines/mkldnn/graph/layers/internal/graph_split_test.cpp [hunk 6]
```diff
@@ -230,27 +230,27 @@ INSTANTIATE_TEST_CASE_P(
                 split_test_params {
                         {1, 24, 2, 5},
                         {{1, 16, 2, 5}, {1, 8, 2, 5}},
-                        1, 5, MKLDNNPlugin::impl_desc_type::ref, {MKLDNNPlugin::impl_desc_type::ref}
```
Comment (a-sidorova) [inline inference-engine/tests_deprecated/unit/engines/mkldnn/graph/layers/internal/graph_split_test.cpp:233]: Comment for whole file:
// Copyright (C) 2018-2021 Intel Corporation

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/split.cpp [hunk 7]
```diff
@@ -91,8 +91,11 @@ const auto planar_5D_ref = CPUSpecificParams{{ncdhw}, {ncdhw}, {"ref"}, "ref"};
 const auto planar_4D = CPUSpecificParams{{nchw}, {nchw}, {}, "unknown"};
 const auto planar_5D = CPUSpecificParams{{ncdhw}, {ncdhw}, {}, "unknown"};
 
-const auto planarChannels_4D = CPUSpecificParams{{nhwc}, {nhwc}, {}, "ref"};
-const auto planarChannels_5D = CPUSpecificParams{{ndhwc}, {ndhwc}, {}, "ref"};
+const auto perChannels_4D = CPUSpecificParams{{nhwc}, {nhwc}, {}, "ref"};
```
[the remark this answers is not in the document]
Comment (maxnick) [reply inference-engine/tests/functional/plugin/cpu/single_layer_tests/split.cpp:94]: Fixed. Thanks!

File: inference-engine/tests_deprecated/unit/engines/mkldnn/graph/layers/internal/graph_split_test.cpp (hunk 6 above)
Comment (maxnick) [reply inference-engine/tests_deprecated/unit/engines/mkldnn/graph/layers/internal/graph_split_test.cpp:233]: Fixed. Thanks!

**Review by a-sidorova — APPROVED** [review]

**Comment by maxnick** [comment]
@a-sidorova Thank you!
@dmitry-gorokhov Could you please review?

**Comment by dmitry-gorokhov** [comment]
@maxnick Please rebase on latest master. I think we also should include fixes for issue #47383 in this PR

**Comment by maxnick** [comment]
Work in progress.

**Comment by maxnick** [comment]
> @maxnick Please rebase on latest master. I think we also should include fixes for issue #47383 in this PR

Done

**Review by alexey-varyzgin — APPROVED** [review]

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**